omod/src/main/java/org/openmrs/module/auditlogweb/rest/AuditLogRestController.java

```diff
@@ -0,0 +1,129 @@
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.auditlogweb.rest;
+
+import lombok.RequiredArgsConstructor;
+import org.openmrs.User;
+import org.openmrs.annotation.Authorized;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.auditlogweb.api.AuditService;
+import org.openmrs.module.auditlogweb.api.dto.AuditLogDetailDTO;
+import org.openmrs.module.auditlogweb.api.dto.AuditLogResponseDto;
+import org.openmrs.module.auditlogweb.api.dto.RestAuditLogDto;
+import org.openmrs.module.auditlogweb.api.utils.AuditLogConstants;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * REST controller for exposing audit log entries via the OpenMRS REST API.
+ * <p>
+ * Provides endpoints to fetch audit logs with optional filtering by user, date range,
+ * and entity type. Supports pagination.
+ * </p>
+ * <p>
+ * Security: Access to the logs is controlled by the {@link AuditLogConstants#VIEW_AUDIT_LOGS} privilege.
+ * </p>
+ *
+ * <p>
+ * Compatible with OpenMRS Platform 2.7.0 and later.
+ * </p>
+ */
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/rest/" + RestConstants.VERSION_1 + "/auditlogs")
+public class AuditLogRestController {
+
+    private final AuditService auditService;
+
+    /**
+     * REST controller for exposing audit log entries via the OpenMRS REST API.
+     * <p>
+     * Provides endpoints to fetch audit logs with optional filtering by:
+     * - user ID or username
+     * - date range (startDate/endDate in "dd/MM/yyyy" format)
+     * - entity type
+     * </p>
+     * <p>
+     * Supports pagination via 'page' (0-based) and 'size' parameters.
+     * </p>
+     * <p>
+     * Security: Access to the logs is controlled by the {@link AuditLogConstants#VIEW_AUDIT_LOGS} privilege.
+     * </p>
+     */
+    @GetMapping
+    public AuditLogResponseDto getAuditLogs(
+            @RequestParam(defaultValue = "0") int page,
+            @RequestParam(defaultValue = "20") int size,
+            @RequestParam(required = false) Integer userId,
+            @RequestParam(required = false) String username,
+            @RequestParam(required = false) String startDate,
+            @RequestParam(required = false) String endDate,
+            @RequestParam(required = false) String entityType
+    ) {
+        if (page < 0) page = 0;
+        if (size <= 0) size = 20;
+
+        Integer effectiveUserId = resolveUserId(userId, username);
+        Date start = parseDate(startDate);
+        Date end = parseDate(endDate);
```

Comment:
These just throw a generic RuntimeException for parse errors. Shouldn't they return a proper HTTP 400 with a meaningful error message?